### PR TITLE
cargo/manifests: allow user to use native-tls if `ring` is unsupported

### PR DESCRIPTION
**Pull request by liushuyu**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [X] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [X] Have you created/updated the relevant documentation page(s)?

-----------

This pull request adds an option to specify the TLS backend. On some platforms like MIPS/PowerPC/RISC-V, where `ring` is unusable, an alternative could be used so that Zola is usable on such platforms.

### CODE REVIEW

File: docs/content/documentation/getting-started/installation.md [hunk 1]
```diff
@@ -155,6 +155,12 @@ From a terminal, you can now run the following command:
 $ cargo build --release
 ```
 
+If you encountered compilation errors like `error: failed to run custom build command for 'ring v0.16.20'`, you can try the command below instead:
+
+```sh
+$ cargo build --release --no-default-features --features=native-tls
```
Comment (Keats) [inline docs/content/documentation/getting-started/installation.md:161]: can we add that command to CI to make sure it runs well? Without the `--release` argument of course

Comment (liushuyu) [reply docs/content/documentation/getting-started/installation.md:161]: > can we add that command to CI to make sure it runs well? Without the `--release` argument of course

Thank you for your reply. I will add a new CI job to test this case.

Comment (Keats) [reply docs/content/documentation/getting-started/installation.md:161]: I think just a step is fine, no need for a new job

Comment (liushuyu) [reply docs/content/documentation/getting-started/installation.md:161]: Done. And all the CI tests seem to be passing.

**Review by Keats — APPROVED** [review]
Thanks!

**Pull request merged**